File: sem2/Controllers/HomeController.cs
using System.Diagnostics;
using DomainModels;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using sem2_FSharp.ViewModels;

//F# version exists, currently doesn't work. TODO fix F# version
namespace sem2.Controllers
{
    public class HomeController : Controller
    {
        private readonly ApplicationContext _dbContext;
        private readonly ILogger<HomeController> _logger;

        public HomeController(ILogger<HomeController> logger, ApplicationContext dbContext)
        {
            _logger = logger;
            _dbContext = dbContext;
        }

        public IActionResult Index()
        {
            return View();
        }

        public IActionResult Privacy()
        {
            return View();
        }
        
        [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
        public IActionResult Error()
        {
            return View(new ErrorViewModel {RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier});
        }
    }
}

File: sem2_FSharp/Controllers/HomeController.fs
namespace sem2_FSharp.Controllers

open System.Diagnostics
open DomainModels
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging
open sem2_FSharp.ViewModels
open sem2_FSharp.NullCoalescing

type HomeController(dbContext: ApplicationContext, logger: ILogger<HomeController>) =
    inherit Controller()
    
    member private this.dbContext = dbContext
    member private this.logger = logger
    
    
    member public this.Index = this.View
    
    member public this.Privacy = this.View
    
    [<ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)>]
    member public this.Error() =
        let activityCurrent = if Activity.Current = null then null else Activity.Current.Id
        let requestId = nullCoalesce activityCurrent this.HttpContext.TraceIdentifier
        this.View(ErrorViewModel(requestId))
    